tests/transactional/enable_selinux.pm

```diff
@@ -49,18 +72,28 @@ sub run {
         }
     }
 
-    assert_script_run('selinuxenabled');
-    validate_script_output("getenforce", sub { m/Enforcing/ });
-    validate_script_output("sestatus", sub { m/Current mode:.*enforcing/ });
-    validate_script_output("sestatus", sub { m/Mode from config file:.*enforcing/ });
-    record_info('SELinux', script_output('sestatus'));
-    record_info('Audit report', script_output('aureport'));
-    record_info('Audit denials', script_output('aureport -a', proceed_on_failure => 1));
-    upload_logs($audit_log);
+    check_enforcing;
+
+    # disable and re-enable SELinux
+    assert_script_run "sed -i -e 's/^SELINUX=.*/SELINUX=disabled/g' /etc/selinux/config";
+    # See "Note: Relabeling your system after switching from the disabled mode" section on:
+    # https://documentation.suse.com/sle-micro/5.4/html/SLE-Micro-all/cha-selinux-slemicro.html
+    assert_script_run "touch /etc/selinux/.autorelabel";
+    process_reboot(trigger => 1);
+    check_disabled;
+    trup_call('setup-selinux');
+    process_reboot(trigger => 1);
+    check_enforcing;
+    assert_script_run "test -f /etc/selinux/.relabelled";
 }
 
 sub test_flags {
     return {fatal => 1, milestone => 1};
 }
 
+sub post_fail_hook {
+    my $audit_log = "/var/log/audit/audit.log";
+    upload_logs($audit_log, log_name => $audit_log . ".txt");
```

Comment:
Does it mean that we don't need audit log when test passes? That's change from previous version.